### ISSUE TITLE
Rule `node/no-unpublished-import` should not be enabled for private packages

### ISSUE THREAD

**Issue by josokinas** [mysticatea/eslint-plugin-node#77]
If `package.json` file has "private" set to *true*, it will restrict the package from being published to the registry. Therefor having `node/no-unpublished-import` rule does not make sense.

**Comment by ClementParis016** [comment]
I agree, and the same observation applies to `node/no-unpublished-require`.
Happy to make a PR if this should be addressed ;)

**Comment by pranshu792** [comment]
Is this still in discussion ?

**Comment by mysticatea** [comment]
At least, I'm using those rules for private packages as well.

**Comment by edg2s** [comment]
Could be an option instead?

**Comment by constgen** [comment]
It is so obvious to me that I can't wait when this will be done. @mysticatea can you show examples of mistakes that you want to prevent with this rule in a private package?

**Comment by uhrohraggy** [comment]
Can we merge this in yet? Has been 5 years since this was opened?

**Comment by angelayanpan** [comment]
can this issue be closed now?

**Comment by voxpelli** [comment]
It has been merged into the maintained for, `eslint-plugin-n`: https://github.com/eslint-community/eslint-plugin-n/pull/57
